src/interfaces/chain.cpp

```diff
@@ -348,13 +348,11 @@ class ChainImpl : public Chain
     {
         return MakeUnique<NotificationsHandlerImpl>(*this, notifications);
     }
-    void waitForNotificationsIfNewBlocksConnected(const uint256& old_tip) override
+    void waitForNotificationsIfTipChanged(const uint256& old_tip) override
     {
         if (!old_tip.IsNull()) {
             LOCK(::cs_main);
             if (old_tip == ::ChainActive().Tip()->GetBlockHash()) return;
-            CBlockIndex* block = LookupBlockIndex(old_tip);
```

Comment:
Updated commit message, due to the switch to height tracking based on callbacks, if we don't wait for the whole callbacks queue being drawn because the last block hash processed by the wallet may be a descendant of the current tip, we may miss to process BlockDisconnected and so get tx deptch inconsistencies.